corehq/apps/reports/filters/users.py

```diff
@@ -266,6 +266,10 @@ def selected_location_ids(mobile_user_and_group_slugs):
     def show_all_mobile_workers(mobile_user_and_group_slugs):
         return 't__0' in mobile_user_and_group_slugs
 
+    @staticmethod
+    def all_users_selected(mobile_user_and_group_slugs):
+        return not any(mobile_user_and_group_slugs)
```

Comment:
I'd also reference this in `user_es_query` to show that the logic is consistent.  Reports which operate on users will want to get the full list of users, but those which operate on forms or cases will instead want to check this and skip the user-based filtering.  Using the same helper method in both instances will show that they're doing the same sort of thing.

Reply:
Should have referenced this everywhere the first time around, my bad
https://github.com/dimagi/commcare-hq/pull/21923/commits/f235200b22b6d08dfd63fe1a200341362d4fc21c